src/rpc/net.cpp

```diff
@@ -511,7 +511,30 @@ static UniValue getnetworkinfo(const JSONRPCRequest& request)
     obj.pushKV("warnings",       GetWarnings("statusbar"));
     return obj;
 }
-
+static UniValue getversioninfo(const JSONRPCRequest& request){
+    RPCHelpMan{"getversioninfo",
+               "Returns an object containing various state info regarding P2P networking.\n",
+               {},
+               RPCResult{
+                       "{\n"
+                       "  \"version\": xxxxx,                      (numeric) the server version\n"
+                       "  \"subversion\": \"/Satoshi:x.x.x/\",     (string) the server subversion string\n"
+                       "  \"protocolversion\": xxxxx,              (numeric) the protocol version\n"
+                       "}\n"                                                                                                                                                                                                                                                      "  \"warnings\": \"...\"                    (string) any network and blockchain warnings\n"
+                                                                                                                                                                                                                                                                                      "}\n"
+               },
+               RPCExamples{
+                       HelpExampleCli("getversioninfo", "")
+                       + HelpExampleRpc("getversioninfo", "")
+               },
+    }.Check(request);
+    LOCK(cs_main);
```

Comment:
Don't need to lock the chain state here.

Reply:
Okay